File: src/TodoServer/client-app/src/entities/todo-item.ts
import { User } from './user'

export class TodoItem {
  id: string
  title: string
  message: string
  status: number
  type: number
  createByUser: User
  assignToUser: User

  constructor(){
    this.id = "00000000-0000-0000-0000-000000000000"
    this.title = ""
    this.message = ""
    this.status = 0
    this.type = 0
    this.createByUser = new User()
    this.assignToUser =  new User()
  }
}

File: src/TodoBoard.Core/Entities/TodoItem.cs
using System;

namespace TodoBoard.Core.Entities
{
  public class TodoItem
  {
    public Guid Id { get; set; }

    public string Title { get; set; }

    public string Message { get; set; }

    public TodoStatus Status { get; set; }

    public DateTime CreateDate { get; set; }

    public DateTime ChangeDate { get; set; }

    //public Guid CreateByUserId { get; set; }

    public User CreateByUser { get; set; }

    //public Guid AssignToUserId { get; set; }

    public User AssignToUser { get; set; }
  }

  public enum TodoStatus : int
  {
    Unknown = 0,
    New = 1,
    InWork = 2,
    Finished = 3
  }
}
